forem/forem
# Remove “Shield” icon in nav for Trusted Users - except admins and tag-moderators

## What type of PR is this? (check all applicable)


## Description

Removing the shield (/mod) icon from the site header for trusted users

![Image](https://user-images.githubusercontent.com/896728/200986752-6bd0b13d-f743-4091-9d2d-ff06963d8825.png)

We can still quietly allow access to `/mod` for the time being — so **no auth changes**, but want to remove the nav icon.

### Definition of Done

- [ ]  Trusted users are no longer able to see the “shield” icon
- [ ]  The icon remains visible to Tag Mods, Admins, superadmins

## Related Tickets & Documents



- Related Issue #18714
- Closes #18714

## QA Instructions, Screenshots, Recordings

**How to test?**
1. Create and sign in as "Admin" user and see the shield icon is visible in nav-bar.
2. Create a new user, assign a role as "Super Admin", visit home page again and see the shield icon is visible.
3. Create a new user, visit home page and see the shield icon is not visible
4. Create anew user, assign some tags to them for moderation, visit home page and see the shield icon is visible.

**How to assign roles?**
1. Visit http://localhost:3000/admin/member_manager/users
2. Select the user and click on the overflow icon (vertical 3-dots)
3. Click on "Assign role"

**How to add tags for moderation?**
1. Visit http://localhost:3000/admin/member_manager/users
2. Click on user name for which tags needs to be added
3. Note the user id from this page, mentioned in URL too.
4. Visit http://localhost:3000/admin/content_manager/tags
5. Click on any tag
6. Add the user id and click "Add Moderator"

### Output / Screenshot

**Case 1**:
- "moderator_for_tags": []
- "admin": false
- **Result** - No shield-icon is visible
<img width="1512" alt="Screenshot 2022-12-05 at 1 19 53 PM" src="https://user-images.githubusercontent.com/9396084/205626480-28bf32ae-792b-4068-9125-adb4545f202c.png">

**Case 2**:
- "moderator_for_tags": ["discuss"]
- "admin": false
- **Result** -Shield-icon is visible
<img width="1512" alt="Screenshot 2022-12-05 at 1 21 08 PM" src="https://user-images.githubusercontent.com/9396084/205626658-abf3cf4b-3cbc-4e1c-9ea6-ba840965179d.png">

**Case 3**:
- "moderator_for_tags": ["discuss"]
- "admin": true
- **Result** - Shield-icon is visible
<img width="1512" alt="Screenshot 2022-12-05 at 1 22 10 PM" src="https://user-images.githubusercontent.com/9396084/205626867-52989421-879b-4645-b546-d089f660cd93.png">

**Case 4**:
- "moderator_for_tags": []
- "admin": true
- **Result** - Shield-icon is visible
<img width="1512" alt="Screenshot 2022-12-05 at 1 22 50 PM" src="https://user-images.githubusercontent.com/9396084/205626981-caf43763-2b9d-4736-80bc-f0da3ddbfad0.png">

**Case 5**:
- Shield-icon is not visible
- Moderator Center is visible in menu
- Trusted user
<img width="1512" alt="Screenshot 2022-12-05 at 1 23 29 PM" src="https://user-images.githubusercontent.com/9396084/205627152-b13cd67b-1fd7-43a5-b7c9-0cae3b2029ef.png">


### UI accessibility concerns?

NA

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
